python/cudf/cudf/tests/test_hdfs.py

```diff
@@ -0,0 +1,204 @@
+import os
+from io import BytesIO
+
+import fastavro as fa
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+import pyarrow.orc as orc
+import pytest
+
+import cudf
+from cudf.tests.utils import assert_eq
+
+if not os.environ.get("RUN_HDFS_TESTS"):
+    pytestmark = pytest.mark.skip("Env not configured to run HDFS tests")
+
+
+basedir = "/tmp/test-hdfs"
+host = "localhost"  # hadoop hostname
+port = 8020  # hadoop rpc port
+
+
+@pytest.fixture
+def hdfs(scope="module"):
+    # Default Rpc port can be 8020/9000 depending on the hdfs config
+    fs = pa.hdfs.connect(host=host, port=port)
+    try:
+        if not fs.exists(basedir):
+            fs.mkdir(basedir)
+    except pa.lib.ArrowIOError:
+        pytest.skip("hdfs config probably incorrect")
+
+    return fs
+
+
+@pytest.fixture
+def pdf(scope="module"):
+    df = pd.DataFrame()
+    df["Integer"] = np.array([2345, 11987, 9027, 9027])
+    df["Float"] = np.array([9.001, 8.343, 6, 2.781])
+    df["Integer2"] = np.array([2345, 106, 2088, 789277])
+    df["String"] = np.array(["Alpha", "Beta", "Gamma", "Delta"])
+    df["Boolean"] = np.array([True, False, True, False])
+    return df
+
+
+@pytest.mark.parametrize("test_url", [False, True])
+def test_csv(tmpdir, pdf, hdfs, test_url):
+    fname = tmpdir.mkdir("csv").join("file.csv")
+    # Write to local file system
+    pdf.to_csv(fname)
+    # Read from local file system as buffer
+    with open(fname, mode="rb") as f:
+        buffer = BytesIO(f.read())
+    # Write to hdfs
+    hdfs.upload(basedir + "/file.csv", buffer)
+
+    if test_url:
+        hd_fpath = "hdfs://{}:{}{}/file.csv".format(host, port, basedir)
+    else:
+        hd_fpath = "hdfs://{}/file.csv".format(basedir)
+
+    got = cudf.read_csv(hd_fpath)
+
+    # Read pandas from byte buffer
+    with hdfs.open(basedir + "/file.csv") as f:
+        expect = pd.read_csv(f)
+
+    assert_eq(expect, got)
```

Comment:
Would be good to test the csv writer as well.

Reply:
The writer does not support writing to remote filesystems at the moment. I could change the pandas `to_csv` call to a `cudf.to_csv` call. Upload that file to HDFS and then compare against that in the tests if that would be preferred.

Reply:
Nope that's fine. We should probably raise a not implemented error in the case a remote filesystem is passed on the `to_csv` call then and write a test to ensure it's raising a NotImplementedError properly.

Reply:
Great. I'll add the `to_csv` tests. Just as an update, right now we raise an error along the lines of `could not open file`. A `notImplementedError` would not be raised since the writers don't really use fsspec or these methods and do not have any idea about whether the filesystem is remote or not. I do plan on raising an issue and was planning on tackling this in a followup pr.

Reply:
👍 that works for me. Thanks!

Reply:
Done